service/initialise.go

```diff
@@ -45,6 +48,27 @@ func (e *ExternalServiceList) GetKafkaConsumer(ctx context.Context, cfg *config.
 	return consumer, nil
 }
 
+func (e *ExternalServiceList) GetCantabularClient(ctx context.Context, cfg *config.Config) CantabularClient {
+	return cantabular.NewClient(
+		dphttp.NewClient(),
+		cantabular.Config{
+			Host: cfg.CantabularURL,
+		},
+	)
+}
+
+func (e *ExternalServiceList) GetDatasetAPIClient(ctx context.Context, cfg *config.Config) DatasetAPIClient {
+	return dataset.NewAPIClient(cfg.DatasetAPIURL)
+}
+
+func (e *ExternalServiceList) GetS3Client(ctx context.Context, cfg *config.Config) (S3Client, error) {
+	s3Client, err := dps3.NewClient(cfg.AWSRegion, cfg.UploadBucketName, true)
+	if err != nil {
+		return nil, err
```

Comment:
When returning errors we always want to a line of  context describing what has just failed. For example here we'd return
`return fmt.Errorf("failed to create S3 Client: %w", err)`
This means when we get the error at the end we can see exactly what is that's failed. Often the core errors can be quite abstract, for example `failed to connect`, without knowing what we're trying to connect to it's much harder to debug the error.